documentation/release_notes.rst

```diff
@@ -72,7 +125,7 @@ New Features
 ------------
 -  Added support of parallel, vector and DPC++ execution policies for the following algorithms: ``shift_left``, ``shift_right``.
 -  Added the range-based versions of the following algorithms: ``sort``, ``stable_sort``, ``merge``.
--  Added non-blocking versions of the following algorithms: ``copy``, ``fill``, ``for_each``, ``reduce``, ``sort``, ``transform``, ``transform_reduce``.
+-  Added asynchronous versions of the following algorithms: ``copy``, ``fill``, ``for_each``, ``reduce``, ``sort``, ``transform``, ``transform_reduce``.
```

Comment:
I'd rephrase to "Added experimental asynchronous algorithms: copy_async, fill_async, ...", also removing the "with suffix _async" in the next sentence.